src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java

```diff
@@ -28,20 +30,114 @@
-            LogUtil.logPrintfError("Please login");
+    public boolean hasAuthorization(String obj, String act) {
+        if (isLogin()) {
             return false;
         }
         Subject subject = SecurityUtils.getSubject();
         return e.enforce(subject.getPrincipal().toString(), obj, act);
     }
+
+    /**
+     * hasAuthorizations determines whether the user currently logged on has a permission of the object.
+     *
+     * @param dom the domain the roles belong to.
+     * @param obj the object (usually means resources).
+     * @param act the action (usually means the operation that the user performs on the resource).
+     * @return whether the current login user has the permission.
+     */
+    public boolean hasAuthorization(String dom, String obj, String act) {
+        if (isLogin()) {
+            return false;
+        }
+        Subject subject = SecurityUtils.getSubject();
+        return e.enforce(subject.getPrincipal().toString(), dom, obj, act);
+    }
+
+
+    /**
+     * hasAuthorization determines whether the user currently logged on has permissions.
+     * Returns false if the user or role already has the permission (aka not affected).
+     *
+     * @param permission the permission, usually be (obj, act). It is actually the rule without the subject.
+     * @return succeeds or not.
+     */
+    public boolean hasPermission(String... permission) {
+        if (isLogin()) {
+            return false;
+        }
+        Subject subject = SecurityUtils.getSubject();
+        return e.hasPermissionForUser(subject.getPrincipal().toString(), permission);
+    }
+
+    /**
+     * hasAuthorization determines whether the user currently logged on has permissions.
+     * Returns false if the user or role already has the permission (aka not affected).
+     *
+     * @param permission the permission, usually be (obj, act). It is actually the rule without the subject.
+     * @return succeeds or not.
+     */
+    public boolean hasPermission(List<String> permission) {
+        if (isLogin()) {
+            return false;
+        }
+        Subject subject = SecurityUtils.getSubject();
+        return e.hasPermissionForUser(subject.getPrincipal().toString(), permission);
+    }
+
+    /**
+     * hasAuthorization determines whether the user currently logged on has permissions.
+     * Returns false if the user or role already has the permission (aka not affected).
+     *
+     * @param role the role.
+     * @return succeeds or not.
+     */
+    public boolean hasRole(String role) {
+        if (isLogin()) {
+            return false;
+        }
+        Subject subject = SecurityUtils.getSubject();
+        return e.hasRoleForUser(subject.getPrincipal().toString(), role);
+    }
+
+
+    /**
+     * isLogin judge whether there is an user currently logged in.
+     *
+     * @return whether there is an user currently logged in.
+     */
+    private boolean isLogin() {
```

Comment:
`isLogin` is not correct English. Should be `isLoggedIn`

Reply:
Ok,I will take your advice.